pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1141,6 +1143,7 @@ func (nc *Controller) tryUpdateNodeHealth(ctx context.Context, node *v1.Node) (t
 					currentCondition.Reason = "NodeStatusUnknown"
 					currentCondition.Message = "Kubelet stopped posting node status."
 					currentCondition.LastTransitionTime = nowTimestamp
+					markPodsNotReady = true
```

Comment:
What if the condition changed from Ready -> NotReady. Shouldn't we also want to markPodsNotReady=true?

I would imagine the cases we want markPodsNotReady=true is if we from ready=true to ready=false or ready=Unknown, i.e:

Ready Condition (true) -> Ready Condition (false)
Ready Condition (true) -> Ready Condition (Unknown)

Reply:
This is a good option, and # 112791 is modified in this way, but it can change existing behavior and cause a lot of UT failures. 

According to the unit test, false-> Unknow is undefined, so I'm just defined this behavior, which there's no impact on the original logic.

Marking pod NotReady makes sense when node `false`, because it cuts off network traffic faster. But `false` is not as serious as `Unknown`, the container will probably survive and the network will still work. At `false` node , the HTTP readiness probe is usually still working and can report the `Ready` status of the pod (Exec Readiness probe may does not work, but this can be a known limitation)

Reply:
IIRC, **only** Node in *Ready Condition (Unknown)* state, node-lifecycle-controller will mark Pods not-ready. And Node in *Ready Condition (False)* state, node-lifecycle-controller will **NOT** mark Pods not-ready.

Based on that, this PR codes could be more simpler and clearer?